native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,621 @@
+            for (ChildNodeEntry child : temp) {
+                if (child.name().equals("metadata")) {
+                    if (child.node().isEmpty()) {
+                        continue;
+                    }
+                    MetadataNode metaD = (MetadataNode) child.node().get();
+                    NodeList<AnnotationNode> annotations = metaD.annotations();
+                    for (AnnotationNode annotation : annotations) {
+                        Node annotReference = annotation.annotReference();
+                        if (annotReference.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                            continue;
+                        }
+                        QualifiedNameReferenceNode qualifiedNameRef =
+                                    (QualifiedNameReferenceNode) annotReference;
+                        if (qualifiedNameRef.identifier().text().equals("Entity") && qualifiedNameRef
+                                .modulePrefix().text().equals(BalFileConstants.PERSIST) &&
+                                annotation.annotValue().isPresent()) {
+                            index += 1;
+                            for (MappingFieldNode fieldNode : annotation.annotValue().get().fields()) {
+                                if (fieldNode.kind() != SyntaxKind.SPECIFIC_FIELD) {
+                                    continue;
+                                }
+                                SpecificFieldNode specificField = (SpecificFieldNode) fieldNode;
+                                if (specificField.fieldName().kind() != SyntaxKind.IDENTIFIER_TOKEN ||
+                                        specificField.valueExpr().isEmpty()) {
+                                    continue;
+                                }
+                                ExpressionNode valueNode =
+                                        specificField.valueExpr().get();
+                                if (valueNode instanceof ListConstructorExpressionNode
+                                        && ((IdentifierToken) specificField.fieldName()).text()
+                                        .equals(BalFileConstants.KEY)) {
+                                    keys = new ArrayList<>();
+                                    Iterator<Node> listIterator = ((ListConstructorExpressionNode) valueNode)
+                                            .expressions().iterator();
+                                    count = 0;
+                                    while (listIterator.hasNext()) {
+                                        keys.add(count, listIterator.next().toString());
+                                        count += 1;
+                                    }
+                                } else if (valueNode instanceof BasicLiteralNode &&
+                                        ((IdentifierToken) specificField.fieldName()).text()
+                                                .equals("tableName")) {
```

Comment:
```suggestion
                                } else if (((SpecificFieldNode) fieldNode).fieldName().toString().trim().equals("tableName")) {
```

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)